plugins/module_utils/nutanix_api_client.py

```diff
@@ -172,3 +207,133 @@ def delete_image(image_uuid, client):
     response = client.request(
         api_endpoint="v3/images/{0}".format(image_uuid), method="DELETE", data=None)
     return response.json()["status"]["execution_context"]["task_uuid"]
+
+
+def get_cluster_uuid(cluster_name, client):
+    length = 250
+    offset = 0
+    total_matches = 99999
+    cluster_uuid = []
+    while offset < total_matches:
+        filter = {"filter": "name=={0}".format(
+            cluster_name), "length": length, "offset": offset}
+        cluster_list = list_entities('clusters', filter, client)
+        for cluster in cluster_list["entities"]:
+            if cluster["status"]["name"] == cluster_name:
+                cluster_uuid.append(cluster["metadata"]["uuid"])
+
+        total_matches = cluster_list["metadata"]["total_matches"]
+        offset += length
+
+    return cluster_uuid
+
+
+def get_subnet_uuid(subnet_name, client):
+    length = 250
+    offset = 0
+    total_matches = 99999
+    subnet_uuid = []
+    while offset < total_matches:
+        filter = {"filter": "name=={0}".format(
+            subnet_name), "length": length, "offset": offset}
+        subnet_list = list_entities('subnets', filter, client)
+        for subnet in subnet_list["entities"]:
+            if subnet["status"]["name"] == subnet_name:
+                subnet_uuid.append(subnet["metadata"]["uuid"])
+
+        total_matches = subnet_list["metadata"]["total_matches"]
+        offset += length
+
+    return subnet_uuid
+
+
+def groups_call(filter, client):
+    sc_list_response = client.request(
+        api_endpoint="v3/groups", method="POST", data=json.dumps(filter))
+    return sc_list_response.json()
+
+
+def get_cluster_storage_container_map(storage_container_name, client):
```

Comment:
Please add doc string for function which will explain about purpose, in args, return value etc